### ISSUE TITLE
nightly version fails: invalid version 3 on git_proxy_options; class=Invalid (3)

### ISSUE THREAD

**Issue by yurivict** [rust-lang/rust#63476]
On FreeBSD 12 amd64 rust-nightly-1.38.0.20190811_1 fails:
```
$ cargo build
    Updating crates.io index
error: failed to load source for a dependency on `image`

Caused by:
  Unable to update registry `https://github.com/rust-lang/crates.io-index`

Caused by:
  failed to fetch `https://github.com/rust-lang/crates.io-index`

Caused by:
  invalid version 3 on git_proxy_options; class=Invalid (3)
```

Project: https://github.com/Twinklebear/bspline

**Comment by jonas-schievink** [comment]
This looks like a packaging issue with libgit2, have you reported this to FreeBSD?

**Comment by yurivict** [comment]
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=239785

**Comment by jbeich** [comment]
> This looks like a packaging issue with libgit2

Our libgit2 package didn't change since 2019-03-31, and old Nightly snapshots work fine:
- 20190811 affected
- 20190806 affected
- 20190723 works fine
- 20190711 works fine

**Comment by jbeich** [comment]
Bisecting cargo led to https://github.com/rust-lang/cargo/pull/7018 but I'm still puzzled given FreeBSD libgit2 package is at 0.28.1.

**Comment by jbeich** [comment]
Updating libgit2 to 0.28.2 doesn't help.

**Comment by semarie** [comment]
I have the same problem with upcoming stable 1.38.0 on OpenBSD, when built with libgit2-0.28.1

**Comment by semarie** [comment]
@jbeich I took your patch for OpenBSD. Thanks for have restored the compatibility

**Comment by Cogitri** [comment]
This has hit Alpine Linux too.

**Comment by cemeyer** [comment]
Can this issue be closed after rust/meta-rust#279 ?
